### PR TITLE
bundle building prototype

### PR DESCRIPTION
**Pull request by wwwtyro**
```sh
$ ls
bundle  fragments  Makefile  tests
$ ./bundle k8s/cdk cni/flannel monitor/elastic
Processing fragment k8s/cdk...
Processing fragment cni/flannel...
Processing fragment monitor/elastic...
Done. Your bundle can be found in /home/rye/src/bundle-canonical-kubernetes/bundle-cdk-flannel-elastic.
```

### CODE REVIEW

**Comment by mbruzek** [comment]
I wanted to comment before I forgot. Can we make the fragments without the revision number in them? The bundle tool that I built looked up the latest revision number from the charmstore api. This did not go far enough and we should make a tool that is channel aware and will take the latest revision from a specific channel.

In the end we want a tool that makes it easier to release bundles. If the fragments do not have versions and we can look that up from an api call that would remove the revision checking from the release process.

File: bundle [hunk 1]
```diff
@@ -54,6 +62,11 @@ def validate_components(fragments):
             raise FragmentMissingComponent('Fragment %s does not contain README.md' % frag)
 
 
+def validate_channel(channel):
```
Comment (mbruzek) [inline bundle]: No docstrings? You are going to add them later right?

File: bundle [hunk 2]
```diff
@@ -67,10 +80,24 @@ def merge(src, dst):
             dst[k] = v
 
 
+def version_bundle(bundle, channel):
```
Comment (mbruzek) [inline bundle]: No docstrings makes me sad

Comment (wwwtyro) [inline bundle]: @mbruzek haha yeah, all added. ;)

**Comment by wwwtyro** [comment]
@mbruzek @chuckbutler @Cynerva @marcoceppi 

I think this is ready for a final look at.

Here's a readme giving the lowdown: https://github.com/juju-solutions/bundle-canonical-kubernetes/tree/rye/buildy/bundler

We should probably get multiple thumbs-up on this - I'm still not sure this is the direction we want to go. One issue is that this is unlikely to extend to generic bundle building. That said, I think it should serve us reasonably well in the near-to-middlin' term for CDK.

**Comment by chuckbutler** [comment]
Will take a look before EOD today, i need to refresh what this is for and how its put together.

File: bundle.yaml [hunk 3]
```diff
@@ -1,72 +1,46 @@
...
-      "gui-y": "850"
-  "flannel":
-    charm: "cs:~containers/flannel-7"
-    annotations:
-      "gui-x": "450"
-      "gui-y": "750"
-  "kubeapi-load-balancer":
-    charm: "cs:~containers/kubeapi-load-balancer-5"
+  easyrsa:
+    charm: cs:~containers/easyrsa-5
```
Comment (mbruzek) [inline bundle.yaml]: I thought we decided to remove the version numbers here?

**Comment by mbruzek** [comment]
I ran the bundle script from this branch and it worked as designed. I also reviewed the python code and do not see any bugs or errors. 

Since comments do not persist from the fragments I would like to add comments to the fragments so people will not try to deploy the incomplete fragments with Juju. This can be another PR in the future.

+1 from my end! Well done.

**Pull request merged**